dask_ml/model_selection/_search.py

```diff
@@ -1193,21 +1177,55 @@ def fit(self, X, y=None, groups=None, **fit_params):
         if scheduler is dask.threaded.get and n_jobs == 1:
             scheduler = dask.local.get_sync
 
-        out = scheduler(dsk, keys, num_workers=n_jobs)
+        if "Client" in type(getattr(scheduler, "__self__", None)).__name__:
+            futures = scheduler(
+                dsk, keys, allow_other_workers=True, num_workers=n_jobs, sync=False
+            )
+            result_map = {}
+            while len(result_map) != len(keys):
+                for f in as_completed(futures):
+                    if f.status == "finished":
+                        result_map[f.key] = f.result()
+                    elif f.status == "error":
+                        f.retry()
```

Comment:
We should log that we're retrying this key.

Reply:
This will be a blocking, synchronous call. Probably not what we want.

Could you instead build up a list of failed keys and call `as_completed` on those? Retrying is important for your original use-case, right?

Reply:
From my understanding we do want to do a synchronous call.  Looking through the code, Dask futures keep the results on the worker until .result() is called (or we use with_results=True in as_completed), if a worker dies before we gather the result we will recompute won't we?